simple/rdm_tagged_peek.c

```diff
@@ -31,95 +31,187 @@
 #include <stdlib.h>
 #include <string.h>
 #include <getopt.h>
+#include <unistd.h>
 
 #include <rdma/fi_errno.h>
 #include <rdma/fi_tagged.h>
 
 #include <shared.h>
 
-static int tagged_peek(uint64_t tag)
+static int wait_for_send_comp(int count)
 {
+	int ret, completions = 0;
 	struct fi_cq_tagged_entry comp;
-	struct fi_msg_tagged msg;
+
+	do {
+		ret = fi_cq_sread(txcq, &comp, 1, NULL, -1);
+		if (ret != 1) {
+			FT_PRINTERR("fi_cq_sread", ret);
+			return ret;
+		}
+		completions++;
+	} while (completions < count);
+
+	return 0;
+}
+
+static int tag_queue_op(uint64_t tag, int recv, uint64_t flags)
+{
 	int ret;
+	struct fi_cq_tagged_entry comp;
+	struct fi_msg_tagged msg = {0};
+	struct fi_cq_err_entry cq_err;
```

Comment:
All fabtests will need to be updated similarly when they bump their version to 1.5.